components/script/dom/htmlscriptelement.rs

```diff
@@ -126,35 +130,48 @@ static SCRIPT_JS_MIMES: StaticStringVec = &[
     "text/x-javascript",
 ];
 
+#[derive(JSTraceable, MallocSizeOf, PartialEq)]
+pub enum ScriptType {
+    Classic,
+    Module,
+}
+
 #[derive(JSTraceable, MallocSizeOf)]
-pub struct ClassicScript {
+pub struct ScriptOrigin {
     text: DOMString,
     url: ServoUrl,
     external: bool,
+    type_: ScriptType,
```

Comment:
```suggestion
    r#type: ScriptType,
```

Reply:
Alternatively,

```suggestion
    script_type: ScriptType,
```

whichever makes the most sense.

Reply:
You could also name it `ScriptKind` instead of  `ScriptType`.